test/e2e/txmgr/txSender.go

```diff
@@ -0,0 +1,146 @@
+package txmgr
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"strings"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/txmgr"
+	t "github.com/omni-network/omni/lib/txmgr"
+	"github.com/omni-network/omni/lib/xchain"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type TxSender struct {
+	txMgr     txmgr.TxManager
+	portal    common.Address
+	abi       *abi.ABI
+	chain     netconf.Chain
+	chainName string
+}
+
+const (
+	gasLimit = 1000000
+	interval = 3
+)
+
+func NewTxSender(
+	ctx context.Context,
+	chain netconf.Chain,
+	rpcClient *ethclient.Client,
+	privateKey ecdsa.PrivateKey,
```

Comment:
this `*ecdsa.PrivateKey` type is always a pointer, see stdlib, lets stick to that pattern